Tests/test_health.gd

```diff
@@ -35,6 +35,22 @@ func test_take_damage_to_block() -> void:
 	assert_eq(_player_health_component.current_block, 5)
 	assert_eq(_player_health_component.current_health, 100)
 
+func test_reset_block_on_round_start() -> void: 
+	var block_amount: float = 10
+	var caster: Entity = _player
+	
+	_player_health_component.add_block(block_amount, caster)
+	PhaseManager.on_phase_changed.emit(GlobalEnums.Phase.PLAYER_ATTACKING, GlobalEnums.Phase.ENEMY_ATTACKING)
+	assert_eq(_player_health_component.current_block, 0)
+
+func test_dont_reset_block_on_enemy_round_start() -> void: 
+	var block_amount: float = 10
+	var caster: Entity = _player
+	
+	_player_health_component.add_block(block_amount, caster)
+	PhaseManager.on_phase_changed.emit(null, GlobalEnums.Phase.PLAYER_ATTACKING)
```

Comment:
maybe emit `GlobalEnums.Phase.ENEMY_ATTACKING` as the new phase ?

Reply:
I did try that, but that did reset the block, which I doubt is a bug with the system, as it doesn't happen when you play the game properly. I think that maybe emiting the GlobalEnums.Phase.ENEMY_ATTACKING will lead to an enemy attack which wil then lead to the player turn starting before the assertion is done

Reply:
that's possible indeed